src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -1276,23 +1275,24 @@ static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * charac
     // Construct the TLV for the additional data
     GVariant * cValue = nullptr;
     CHIP_ERROR err    = CHIP_NO_ERROR;
-    System::PacketBufferTLVWriter writer;
-    TLVWriter innerWriter;
     chip::System::PacketBufferHandle bufferHandle;
 
-    writer.Init(chip::System::PacketBuffer::New());
+    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    size_t serialNumberSize     = 0;
+    uint16_t lifetimeCounter    = 0;
+    bool enableRotatingDeviceId = false;
 
-    err = writer.OpenContainer(AnonymousTag, kTLVType_Structure, innerWriter);
+    err = ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber), serialNumberSize);
     SuccessOrExit(err);
-
-    // Adding the rotating device id to the TLV data
-    err = innerWriter.PutString(ContextTag(kRotatingDeviceIdTag), CHIP_ROTATING_DEVICE_ID);
+    err = ConfigurationMgr().GetLifetimeCounter(lifetimeCounter);
```

Comment:
Everything related to serial/rotating ID must be conditional to `CHIP_ENABLE_ROTATING_DEVICE_ID`. The code paths should not even exist without.

Reply:
updated